lib/v2/magazinelib/latest-magazine.js

```diff
@@ -0,0 +1,72 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { art } = require('@/utils/render');
+const path = require('path');
+
+const host = 'https://magazinelib.com';
+module.exports = async (ctx) => {
+    const query = ctx.params.query;
+    const url = host;
+    const response = await got({
+        method: 'get',
+        url,
+        searchParams: {
+            s: query,
+        },
+    });
+    let subTitle = query;
+    if (subTitle !== undefined) {
+        subTitle = subTitle.replace(/[^a-zA-Z0-9]+/g, ' ').toUpperCase();
+        subTitle = ` - ${subTitle}`;
+    } else {
+        subTitle = '';
+    }
+
+    const data = response.data;
+    const $ = cheerio.load(data);
+
+    let list = $('.g1-collection-item');
+    list = list.filter(function () {
+        const title = $(this).find('.entry-title').text();
+        return title.length !== 0;
+    });
+    list = list
+        .map(function () {
+            return { link: $(this).find('.entry-title a').attr('href') };
+        })
+        .get();
+
+    const items = await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: item.link,
+                });
+                const data = {};
+                const content = cheerio.load(detailResponse.data);
+                data.title = content('.entry-title').text();
+                data.categories = content('p.entry-categories a')
+                    .map((_, el) => $(el).text().trim())
+                    .get();
+                data.imgUrl = content('.vkwpb_img_wrap img').attr('data-lazy-src');
+                data.pubDate = content('.entry-date').attr('datetime');
+                data.downloadLink = content('.vk-att-item > a').attr('href');
+                data.description = art(path.join(__dirname, 'templates/magazine-description.art'), { data });
+                return data;
+            })
+        )
+    );
+    ctx.state.data = {
+        title: `MagazineLib - Latest Magazines${subTitle}`,
+        link: response.url,
+        description: `MagazineLib - Latest Magazines${subTitle}`,
+        item: items.map((item) => ({
+            title: item.title,
+            link: item.downloadLink,
```

Comment:
`link` should point to the site `magazinelib.com` but not external sites.
https://github.com/DIYgod/RSSHub/blob/1dfb37e96b27bf5222c968511e52b2abdaf6acb0/lib/v2/magazinelib/latest-magazine.js#L34-L36